app/main.py

```diff
@@ -1,25 +1,90 @@
 from abc import ABC
+from typing import Union, Type
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: Union["SlideLimitationValidator", "Visitor"],
+            name: str
+    ) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            instance: Union["SlideLimitationValidator", "Visitor"],
+            owner: Union["SlideLimitationValidator", "Visitor"]
```

Comment:
owner is type
Use debuging and just do
print(type(instance))
print(type(owner))